core/src/test/java/com/linecorp/armeria/server/http/HttpServiceTest.java

```diff
@@ -91,12 +104,41 @@ protected void doGet(
                               res.respond(HttpStatus.NO_CONTENT);
                           }
                       }.decorate(LoggingService::new));
+            sb.service(PathMapping.ofPrefix("/dynamic"), new DynamicHttpServiceBuilder()
+                    .add(HttpMethod.POST, "/{var}", (ctx, req) -> {
+                             final DefaultHttpResponse res = new DefaultHttpResponse();
```

Comment:
Instead of using `HttpService`, how about defining a new interface that's more suitable for dynamic HTTP services? For example:

```java
@FunctionalInterface
public interface .... {
    void serve(ServiceRequestContext ctx, HttpRequest req, Map<String, String> params, HttpResponseWriter res);
}
```